### PR TITLE
KYP-625: Handle phone orders in woocommerce plugin

### PR DESCRIPTION
**Pull request by nedick007**
Create email address from phone number if email is not present in the order details.

### CODE REVIEW

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready for review

File: includes/integration.php [hunk 1]
```diff
@@ -346,7 +346,7 @@ public function sync_orders_chunk($specific_order_ids = false){
 
                 try {
                     $order = new WC_Order($order_id);
-                    if(!empty($order_id) && !empty($order) && $this->stringIsPresent(get_post_meta($order_id, '_billing_email', true))){
+                    if(!empty($order_id) && !empty($order) && ($this->stringIsPresent(get_post_meta($order_id, '_billing_email', true)) || $this->stringIsPresent(get_post_meta($order_id, '_billing_phone', true)))){
```
Comment (evgeni-spasov-metrilo) [inline includes/integration.php]: You should change that a bit, so a person can read what is going on in the `if` statement:
```suggestion
                    $has_email = $this->stringIsPresent(get_post_meta($order_id, '_billing_email', true));
                    $has_phone = $this->stringIsPresent(get_post_meta($order_id, '_billing_phone', true));
                    
                    if(!empty($order_id) && !empty($order) && ($has_email || $has_phone)){
```

File: includes/integration.php [hunk 2]
```diff
@@ -799,20 +804,25 @@ public function new_order_event($order_id){
         $order = new WC_Order($order_id);
 
         $call_params = false;
-
-        if(!$this->stringIsPresent(get_post_meta($order_id, '_billing_email', true))) {
+    
+        if(!$this->stringIsPresent(get_post_meta($order_id, '_billing_email', true)) && !$this->stringIsPresent(get_post_meta($order_id, '_billing_phone', true))) {
```
Comment (evgeni-spasov-metrilo) [inline includes/integration.php]: Same here. Get those values in `$email` and `$phone` variables and use them after that also (lines 811 on).

File: includes/integration.php [hunk 3]
```diff
@@ -799,20 +804,25 @@ public function new_order_event($order_id){
...
         $this->identify_call_data = array(
-            'id'        => get_post_meta($order_id, '_billing_email', true),
+            'id'        => $email ? $email : $phone . '@phone_email',
             'params'    => array(
-                        'email' 		=> get_post_meta($order_id, '_billing_email', true),
-                        'first_name' 	=> get_post_meta($order_id, '_billing_first_name', true),
-                        'last_name' 	=> get_post_meta($order_id, '_billing_last_name', true),
-                        'name'			=> get_post_meta($order_id, '_billing_first_name', true) . ' ' . get_post_meta($order_id, '_billing_last_name', true),
-                    )
+                'email' 		=> $email ? $email : $phone . '@phone_email',
```
Comment (evgeni-spasov-metrilo) [inline includes/integration.php]:
```php
'id'        => $email ? $email : $phone . '@phone_email',
...
'email' 		=> $email ? $email : $phone . '@phone_email',
```
those look pretty similar to each other.

**Review by evgeni-spasov-metrilo — CHANGES_REQUESTED** [review]

File: includes/integration.php (hunk 1 above)
Comment (nedick007) [reply includes/integration.php]: Refactored.

File: includes/integration.php (hunk 2 above)
Comment (nedick007) [reply includes/integration.php]: Refactored.

File: includes/integration.php (hunk 3 above)
Comment (nedick007) [reply includes/integration.php]: Refactored.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo bump

**Review by evgeni-spasov-metrilo — APPROVED** [review]

**Pull request merged**